### ISSUE TITLE
It should be possible for integration tests to bring up a drogulus network of N nodes running on localhost

### ISSUE THREAD

**Issue by ntoll** [ntoll/drogulus#26]
For the integration test suite to work properly and fully exercise all the capabilities of target test-mode (see #22) nodes it should be trivial to bring up a "swarm" of nodes running on `localhost`.

There should be some convention for allowing the integration test suite (and users) to inspect and check the logs generated by the nodes in order to confirm the expected behaviour has taken place.

**Comment by HaMster21** [comment]
Sounds like you are shouting `Docker` or `rkt` here :wink:

**Comment by ntoll** [comment]
You're right... there are various options which are worth exploring. I'm already quite familiar with docker, but less so with rkt (and thanks for the pointer). 

It'd be great if it were platform agnostic (which docker sort-of allows). Then again, spinning up a bunch of nodes via many sub-processes (via `Popen` for example) could also work - although I'm unsure how well this would perform.

**Comment by HaMster21** [comment]
[rkt](https://github.com/coreos/rkt) is not yet as stable as Docker as far as I know. But it's the better choice in terms of trust and security in my opinion. rkt make cryptografically sure that images are untouched before running them and they are really authored by the person on the cover of the image. Although it works a little different, it's also able to run Docker images with rkt.

**Comment by kunthar** [comment]
It's PITA for the open source projects to get cloud abilities "without" greedy cloud providers. GCE and Amazon have no special discounts for OS projects. On the other hand, running a dedicated server is really hard and not feasible in long term usage. There are some app centric solutions like Heroku and Bluemix but their pricing policy is not perfectly fit when the number of vms/containers grow as project grows. Also it is impossible to manage internal aspects of your application like networking.

I wish we could have a dedicated OS community to construct sustainable open cloud, based on Openstack for OS projects. But for now best option is either dedicated servers or small cheap VM providers like Digitalocean. :8ball:

**Comment by ntoll** [comment]
@kunthar thanks for the heads up.
